lesson06/task3/server/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+)
+
+var name string
+var address string
+
+func handler(w http.ResponseWriter, r *http.Request) {
+
+	if r.URL.Path != "/" {
+		http.Error(w, "404 not found.", http.StatusNotFound)
+		return
+	}
+
+	switch r.Method {
+	case "GET":
+
+		// GET return HTML page
+		http.ServeFile(w, r, "form.html")
+
+	case "POST":
+
+		if err := r.ParseForm(); err != nil {
+			fmt.Fprintf(w, "ParseForm() err: %v", err)
+			return
+		}
+
+		name = r.FormValue("name")
+
+		address = r.FormValue("address")
+
+		fmt.Fprintf(w, "%s from %s\n", name, address)
+		fmt.Printf("%s from %s\n", name, address)
+
+	default:
+		fmt.Fprintf(w, "Sorry, only GET and POST methods are supported.")
```

Comment:
I think this message is not appropriate for a web server. Better return 404 in this case